### PR TITLE
show namespace for custom strategy bc

### PR DESCRIPTION
**Pull request by juanvallejo**
Related BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1365460

If there is a `bc` using a custom strategy, `oc get all --all-namespaces` does not show namespace for that `bc`.

cc @fabianofranz

### CODE REVIEW

**Comment by juanvallejo** [comment]
[test]

File: test/cmd/basicresources.sh [hunk 1]
```diff
@@ -307,6 +307,11 @@ os::cmd::expect_success 'oc new-app node'
 os::cmd::expect_success_and_text 'oc get all' 'is/test1'
 os::cmd::expect_success_and_not_text 'oc get is' 'is/test1'
 echo "resource printer: ok"
+# Test that resource printer includes namespaces for buildconfigs with custom strategies
+os::cmd::expect_success 'oc create -f examples/sample-app/application-template-custombuild.json'
+os::cmd::expect_failure_and_text 'oc new-app ruby-helloworld-sample' 'deploymentconfig "frontend" created'
+os::cmd::expect_success_and_text 'oc get all --all-namespaces' 'cmd-basicresources[\ ]+bc\/ruby\-sample\-build'
+echo "resource printer namespace handling: ok"
```
Comment (stevekuznetsov) [inline test/cmd/basicresources.sh]: Instead of adding new tests to this suite, let's break out resource printer tests from `basicresources`

**Comment by fabianofranz** [comment]
LGTM [merge]

File: test/cmd/basicresources.sh (hunk 1 above)
Comment (juanvallejo) [inline test/cmd/basicresources.sh]: Will do, I can make a new PR that does this

**Comment by stevekuznetsov** [comment]
@fabianofranz hold on... we need to be treating our tests better. More tests should not go into `basicresources`. There's no point in merging this in if I or Juan will just have to clean it up later.

File: test/cmd/basicresources.sh (hunk 1 above)
Comment (stevekuznetsov) [inline test/cmd/basicresources.sh]: Add it to this one

**Comment by juanvallejo** [comment]
@stevekuznetsov 
> Add it to this one

Done!

File: test/cmd/printer.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${OS_ROOT}/hack/lib/init.sh"
+os::log::stacktrace::install
+trap os::test::junit::reconcile_output EXIT
+
+# Test that resource printer includes resource kind on multiple resources
+os::test::junit::declare_suite_start "cmd/basicresources/get"
```
Comment (stevekuznetsov) [inline test/cmd/printer.sh]: if this is a suite for the resource printer, I think we should rename it that instead of `get`

**Comment by juanvallejo** [comment]
flake re[test]

**Comment by juanvallejo** [comment]
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_integration/4874/console

@openshift-bot, the [last build](https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/7797/) failed from the following flakes:
 - [check](https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_integration/4874/console) failed on: https://github.com/openshift/origin/issues/8027

re[test]

**Pull request merged**